tests/mgbench/benchmark.py

```diff
@@ -175,136 +536,268 @@ def filter_benchmarks(generators, patterns):
 benchmarks = filter_benchmarks(generators, args.benchmarks)
 
 # Run all specified benchmarks.
-for dataset, tests in benchmarks:
+for dataset, queries in benchmarks:
+
+    workload = Workload(args.mixed_workload)
+
+    run_config = {
+        "vendor": args.vendor_name,
+        "condition": "hot" if args.warmup_run else "cold",
+        "workload": workload.name,
+        "workload_config": workload.config,
+    }
+
+    results.set_value("__run_configuration__", value=run_config)
+
     log.init("Preparing", dataset.NAME + "/" + dataset.get_variant(), "dataset")
     dataset.prepare(cache.cache_directory("datasets", dataset.NAME, dataset.get_variant()))
 
-    # Prepare runners and import the dataset.
-    memgraph = runners.Memgraph(
-        args.memgraph_binary,
-        args.temporary_directory,
-        not args.no_properties_on_edges,
-        args.test_system_args,
-    )
-    client = runners.Client(args.client_binary, args.temporary_directory)
-    memgraph.start_preparation()
-    time.sleep(5.0)  # giving enough time to machine manager and all to start up
-    ret = client.execute(file_path=dataset.get_file(), num_workers=args.num_workers_for_import)
-    usage = memgraph.stop()
-
-    # Display import statistics.
-    print()
-    for row in ret:
-        print(
-            "Executed",
-            row["count"],
-            "queries in",
-            row["duration"],
-            "seconds using",
-            row["num_workers"],
-            "workers with a total throughput of",
-            row["throughput"],
-            "queries/second.",
+    # TODO: Create some abstract class for vendors, that will hold this data
+    if args.vendor_name == "neo4j":
+        vendor = runners.Neo4j(
+            args.vendor_binary,
+            args.temporary_directory,
+            args.bolt_port,
+            args.performance_tracking,
+            args.test_system_args,
         )
-    print()
-    print(
-        "The database used",
-        usage["cpu"],
-        "seconds of CPU time and peaked at",
-        usage["memory"] / 1024 / 1024,
-        "MiB of RAM.",
-    )
+    else:
+        vendor = runners.Memgraph(
+            args.vendor_binary,
+            args.temporary_directory,
+            not args.no_properties_on_edges,
+            args.bolt_port,
+            args.performance_tracking,
+            args.test_system_args,
+        )
+
+    client = runners.Client(args.client_binary, args.temporary_directory, args.bolt_port)
```

Comment:
Client should end up as a part of runner or in that additional abstraction (the abstraction is not required)